Modules/Core/Common/src/itkProcessObject.cxx

```diff
@@ -1163,8 +1163,17 @@ ProcessObject::IncrementProgress(float increment)
   // Clamp the value to be between 0 and 1.
   uint32_t integerIncrement = progressFloatToFixed(increment);
 
+  uint32_t oldProgress = m_Progress;
+
   m_Progress.fetch_add(integerIncrement);
 
+  if (oldProgress < m_Progress)
```

Comment:
This check seems tautological: condition should not be met only only if no progress was made, or there are some multiple thread race conditions.

Reply:
The line has been corrected, and a comment added for its reason. Thank you for point it out.